05-lists-and-dictionaries/poetry.py

```diff
@@ -0,0 +1,114 @@
+# Copyright (c) 2016 Ami . All rights reserved
+
+from random import choice
+import re
+
+class SpeechPart(object):
+    def __init__(self, name, count, vocabulary):
+        self.name = name
+        self.count = count
+        self.vocabulary = vocabulary
+
+    def random_words(self):
+        words = []
+
+        while True:
+            if len(words) == self.count:
+                return words
+            else:
+                if len(self.vocabulary) == len(words):
+                    return words
+                selected_word = choice(self.vocabulary)
+                if selected_word not in words:
+                    words.append(selected_word)
+
+        return words
+
+def make_poem():
+    speech_parts = [
+        SpeechPart(
+            name="noun",
+            count=3,
+            vocabulary=[
+                "fossil",
+                "horse",
+                "aardvark",
+                "judge",
+                "chef",
+                "mango",
+                "extrovert",
+                "gorilla"
+            ]
+        ),
+        SpeechPart(
+            name="verb",
+            count=3,
+            vocabulary=[
+                "kicks",
+                "jingles",
+                "bounces",
+                "slurps",
+                "meows",
+                "explodes",
+                "curdles"
+            ]
+        ),
+        SpeechPart(
+            name="adjective",
+            count=3,
+            vocabulary=[
```

Comment:
You either put the whole list on one line or break it, this is a middle state where it's both broken and not, and it's not according to the style guide. I think this will fit in 1 line.
Good: `vocabulary=["a", "b", ...]`
Good:
```
vocabulary=[
    "a",
    "b",
    ...
]
```
Bad:
```
vocabulary=[
    "a", "b", ...
]
```